website/project/model.py

```diff
@@ -2620,3 +2769,226 @@ def to_json(self):
+        return ('<Embargo(parent_registration={0}, initiated_by={1}, '
+                'end_date={2}) with _id {3}>').format(
+            parent_registration,
+            self.initiated_by,
+            self.end_date,
+            self._id
+        )
+
+    @property
+    def embargo_end_date(self):
+        if self.state == 'active':
+            return self.end_date.strftime("%A, %b. %d, %Y")  # e.g. 'Friday, Jan. 01, 2016'
+        return False
+
+    @property
+    def pending_embargo(self):
+        return self.state == Embargo.UNAPPROVED
+
+    # NOTE(hrybacki): Old, private registrations are grandfathered and do not
+    # require to be made public or embargoed. This field differentiates them
+    # from new registrations entering into an embargo field which should not
+    # show up in any search related fields.
+    @property
+    def pending_registration(self):
+        return not self.for_existing_registration and self.pending_embargo
+
+    def disapprove_embargo(self, user, token):
+        """Cancels retraction if user is admin and token verifies."""
+        try:
+            if self.approval_state[user._id]['disapproval_token'] != token:
+                raise InvalidEmbargoDisapprovalToken('Invalid embargo disapproval token provided.')
+        except KeyError:
+            raise PermissionsError('User must be an admin to disapprove embargoing of a registration.')
+
+        self.state = Embargo.CANCELLED
+        parent_registration = Node.find_one(Q('embargo', 'eq', self))
+        # Remove backref to parent project if embargo was for a new registration
+        if not self.for_existing_registration:
+            parent_registration.registered_from = None
+        parent_registration.add_log(
+            action=NodeLog.EMBARGO_CANCELLED,
+            params={
+                'node': parent_registration._id,
+                'embargo_id': self._id,
+            },
+            auth=Auth(user),
+            log_date=datetime.datetime.utcnow(),
+            save=True,
+        )
+        # Delete parent registration if it was created at the time the embargo was initiated
+        if not self.for_existing_registration:
+            parent_registration.is_deleted = True
+            parent_registration.save()
+
+    def approve_embargo(self, user, token):
+        """Add user to approval list if user is admin and token verifies."""
+        try:
+            if self.approval_state[user._id]['approval_token'] != token:
+                raise InvalidEmbargoApprovalToken('Invalid embargo approval token provided.')
+        except KeyError:
+            raise PermissionsError('User must be an admin to disapprove embargoing of a registration.')
+
+        self.approval_state[user._id]['has_approved'] = True
+        if all(val['has_approved'] for val in self.approval_state.values()):
+            self.state = Embargo.ACTIVE
+            parent_registration = Node.find_one(Q('embargo', 'eq', self))
```

Comment:
It should not be possible. The only way it could happen is if someone went into the database, found a registration with an associated Embargo or Retraction, and manually broke the link between them.